#### Chun Shen Lim, Brooke N. Weinstein, Scott W. Roy, Chris M. Brown. (2021). Analysis of fungal genomes reveals commonalities of intron loss or gain and functions in intron-poor species. Mol Biol Evol. DOI: https://doi.org/10.1093/molbev/msab094.

This repository contains the Jupyter notebooks and data for our analyses.
- Jupyter notebooks and timetrees for [phylogenetic independent contrasts analysis](https://github.com/Brookesloci/fungi_intron_paper_2020/tree/master/phylogenetic_independent_contrasts). 
- Jupyter notebooks and [annotation files](https://zenodo.org/record/4273072) for [intron site histories analysis](https://github.com/Brookesloci/fungi_intron_paper_2020/tree/master/intron_site_histories).
-	A Jupyter notebook and datasets for enrichment analysis of [repeat elements](https://github.com/Brookesloci/fungi_intron_paper_2020/tree/master/repeat_elements).
-	Input files for [Malin](http://www.iro.umontreal.ca/~csuros/introns/malin/): sequence alignment files for [1,444 sets of orthologs](https://github.com/Brookesloci/fungi_intron_paper_2020/tree/master/intron_alignment_files), and a [taxa tree](https://github.com/Brookesloci/fungi_intron_paper_2020/tree/master/malin_input_files). 
-	Intermediate files generated using [Malin](http://www.iro.umontreal.ca/~csuros/introns/malin/): [intron table and rate files](https://github.com/Brookesloci/fungi_intron_paper_2020/tree/master/malin_input_files). 
-	[Output files and results](https://github.com/Brookesloci/fungi_intron_paper_2020/tree/master/results_data) for the inference of ancestral introns. 
-	A custom [GTF file of UTR annotation](https://github.com/Brookesloci/fungi_intron_paper_2020/tree/master/annotation_files) for *Saccharomyces cerevisiae*.
